### PR TITLE
[CELEBORN-1340] Congestion control slow start logic should make all interval can be customized and make it smaller

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
### What changes were proposed in this pull request?
In this pr we do two things:

1. Current sleep time is too long for small task, will cause task delay so long, some small task from 30m to 2h
2. Make all interval can be customized

### Why are the changes needed?
Fix performance issue and make it can be customized

### CODE REVIEW

**Comment by AngersZhuuuu** [comment]
ping @onebox-li @waitinfuture

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 1]
```diff
@@ -3372,8 +3375,24 @@ object CelebornConf extends Logging {
...
       .doc(s"The initial sleep time if the current max in flight requests is 0")
-      .timeConf(TimeUnit.MILLISECONDS)
-      .createWithDefaultString("500ms")
+      .timeConf(TimeUnit.MICROSECONDS)
+      .createWithDefaultString("500us")
+
+  val CLIENT_PUSH_SLOW_START_STEP_TIME: ConfigEntry[Long] =
+    buildConf("celeborn.client.push.slowStart.stepTime")
+      .categories("client")
+      .version("0.3.0")
```
Comment (cxzl25) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:3384]: `0.3.0` ?

Comment (AngersZhuuuu) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala]: Changed to 0.5.0

**Review by onebox-li — COMMENTED** [review]
Sorry for the late reply. Orz
LGTM for the variables involved are configurable and decrease the default value. 
But I wonder whether the change of default sleep time from milliseconds to microseconds could be effectively limited for congestion conditions.FYI, https://github.com/apache/celeborn/pull/1637#discussion_r1244944240
And the operating system may not be able to accurately control such a short sleep time.
BTW, migration.md and related UT maybe need to be updated if the default value changed.

**Pull request closed without merging**